api/web/base.py

```diff
@@ -27,6 +27,11 @@ def __init__(self, request=None, response=None): # pylint: disable=super-init-no
         """Set uid, source_site, public_request, and superuser"""
         self.initialize(request, response)
 
+        # If user is attempting to log in through `/login`, ignore Auth here:
+        # In future updates, move login and logout handlers to class that overrides this init
+        if self.request.path == 'api/login':
+            return
```

Comment:
Ah. As an update to my "What was broken without this line?" comment, are these two pieces of code linked? Basically bypass legacy logic, and only `uid` was missing?